frame/nomination-pools/src/lib.rs

```diff
@@ -2117,6 +2373,76 @@ pub mod pallet {
 			ensure!(bonded_pool.can_nominate(&who), Error::<T>::NotNominator);
 			T::Staking::chill(&bonded_pool.bonded_account())
 		}
+
+		/// Set the commission of a pool.
+		///
+		/// The dispatch origin of this call must be signed by the `root` role of the pool. Both a
+		/// commission percentage and a commission payee must be provided in the `current` tuple.
+		/// Where a `current` of `None` is provided, any current commission will be removed.
+		#[pallet::call_index(14)]
+		#[pallet::weight(T::WeightInfo::set_commission())]
+		pub fn set_commission(
+			origin: OriginFor<T>,
+			pool_id: PoolId,
+			new_commission: Option<(Perbill, T::AccountId)>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let mut bonded_pool = BondedPool::<T>::get(pool_id).ok_or(Error::<T>::PoolNotFound)?;
+			ensure!(bonded_pool.can_set_commission(&who), Error::<T>::DoesNotHavePermission);
+
+			bonded_pool.commission.try_update_current(&new_commission)?;
+			bonded_pool.put();
+			Self::deposit_event(Event::<T>::PoolCommissionUpdated {
+				pool_id,
+				current: new_commission,
+			});
+			Ok(())
+		}
+
+		/// Set the maximum commission of a pool.
+		///
+		/// The dispatch origin of this call must be signed by the `root` role of the pool.
```

Comment:
These are mentioned in the comment of `try_update_max`. In the interest of not repeating comments it is only mentioned once there.

Reply:
I wonder if those docs will show up for non-public function. It may be more useful to document the public functions than the internal ones.

Am looking at [this](https://paritytech.github.io/substrate/master/pallet_nomination_pools/pallet/struct.Pallet.html) and you only see the public apis (which makes sense).

Reply:
Added some notes to call comments.